client/app/scripts/superdesk-authoring/metadata/metadata.js

```diff
@@ -27,6 +27,22 @@ function MetadataCtrl($scope, desks, metadata, $filter, privileges, adminPublish
         });
     };
 
+    $scope.$watch('item.publish_schedule_date', function() {
```

Comment:
fyi this could be like `$scope.$watch('item.publish_schedule_date', setPublishScheduleDate);`